CMakeLists.txt

```diff
@@ -48,6 +48,40 @@ if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
   set(MASTER_PROJECT ON)
 endif()
 
+option(USE_VALGRIND "Add extra annotations for better valgrind handling" OFF)
```

Comment:
If we were to introduce these, they should be in another PR
Keep them for now while debugging if you want but remove them before the final merge